dbcon/joblist/tuplehashjoin.cpp

```diff
@@ -407,7 +406,8 @@ void TupleHashJoinStep::smallRunnerFcn(uint32_t index, uint threadID, uint64_t *
             utils::releaseSpinlock(rgdLock);
 
             rgSize = smallRG.getSizeWithStrings();
-            atomicops::atomicAdd(&memUsedByEachJoin[index], rgSize);
+            memUsedByThisJoin += rgSize;
+//            atomicops::atomicAdd(&memUsedByEachJoin[index], rgSize);
```

Comment:
Comment?

Reply:
This is questionable and I am of two minds about it. But inside this function is the logic to determine if a disk join should be entered. Since other threads are also using memory, the switch to diskjoin was undeterministic.

The switch to diskjoin is supposed to be determined by the amount of memory required for the small side dataset. By detaching the count of this memory from the overall query memory count, it becomes deterministic.

If you prefer, I'll switch it back to the way it was.

Reply:
My point was about having this unused statement as the comment in  production code.

Reply:
OK, I'll remove it.